source/common/http/conn_manager_impl.cc

```diff
@@ -278,7 +278,9 @@ RequestDecoder& ConnectionManagerImpl::newStream(ResponseEncoder& response_encod
   // accounting are implemented.
   Buffer::BufferMemoryAccountSharedPtr downstream_request_account;
   if (Runtime::runtimeFeatureEnabled("envoy.test_only.per_stream_buffer_accounting")) {
-    downstream_request_account = std::make_shared<Buffer::BufferMemoryAccountImpl>();
+    // Create account, wiring the stream to use it.
+    auto& buffer_factory = read_callbacks_->connection().dispatcher().getWatermarkFactory();
+    downstream_request_account = buffer_factory.createAccount(&response_encoder.getStream());
```

Comment:
Very dumb question - why are we talking about a downstream request account but all the wiring is happening for the response encoder?

Reply:
The primary owner of the account right is the server-side(e.i. downstream) stream.  Perhaps we should change "request" to "stream" in this variable name.

There's a desire to split the accounts used for the request and response directions, but have both of those accounts associated primarily with the downstream stream.  But that change will happen in a future PR.

Reply:
I've changed it to `downstream_stream_account` as suggested.